src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/GeoIpProcessorFactory.java

```diff
@@ -12,17 +12,17 @@
 import java.util.Map;
 
 public class GeoIpProcessorFactory implements Processor.Factory {
-    private final GeoIpDatabaseProvider geoIpDatabaseProvider;
+    private final IpDatabaseProvider ipDatabaseProvider;
 
-    public GeoIpProcessorFactory(final GeoIpDatabaseProvider geoIpDatabaseProvider) {
-        this.geoIpDatabaseProvider = geoIpDatabaseProvider;
+    public GeoIpProcessorFactory(final IpDatabaseProvider ipDatabaseProvider) {
+        this.ipDatabaseProvider = ipDatabaseProvider;
     }
 
     @Override
     public Processor create(Map<String, Processor.Factory> processorFactories,
                             String tag,
                             String description,
                             Map<String, Object> config) throws Exception {
-        return new GeoIpProcessor.Factory(this.geoIpDatabaseProvider).create(processorFactories, tag, description, config);
+        return new GeoIpProcessor.Factory("geoip", this.ipDatabaseProvider).create(processorFactories, tag, description, config);
```

Comment:
review note: addition to the spike, the `GeoIpProcessor.Factory` requires processor type and taking a look at the ES source, it seems we need to require `geoip` type.